File: trab2/tls-server.js
const tls = require('tls')
const fs = require('fs')
const https = require('https')
const express = require('express')


const options = { 
    key: fs.readFileSync('./tls-cer/server-key.pem'), 
    cert: fs.readFileSync('./tls-cer/server-crt.pem'), 
    ca: fs.readFileSync('./tls-cer/ca-crt.pem'), 
    requestCert: true, 
    rejectUnauthorized: true
}

const configApi = () => {
    const app = express();
  
    authorizationsMiddleware(app);
    flamengoRoutes(app);
  
    return app;
  };














// const server = tls.createServer(options, (socket) => {
//     console.log(socket)

//     console.log('server connected', 
//         socket.authorized ? 'authorized' : 'unauthorized')
    
//     socket.on('error', (error) => {
//         console.log(error)
//     })
        
//     socket.write('welcome!\n')
//     socket.setEncoding('utf8')

//     socket.on('data', (data) => {
//         console.log('datastage')
//         console.log(data)
//         console.log('afterdatastage')
//         if(data=="end\r\n"){
//             socket.end('goodbye\n')
//             server.close()
//         }
//     }).on('error', (err) => {
//         // handle errors here
//         console.log('handle errors here')
//         throw err;
//       })
// })

// server.listen(9443, () => {
//     console.log('server listening port 9443')
// });

File: trab2/tlsServer.py
import ssl
from socket import *

HOSTNAME = 'www.python.org'
IP = '127.0.0.1'
PORT = 4444
ADDRESS = (IP, PORT)

def main():
    server = socket(AF_INET, SOCK_STREAM)
    server.bind(ADDRESS)
    server.listen(2)
    
    print("Rodando...")    
    while True:
        obj, addr = server.accept()
        serverSsl = ssl.wrap_socket(
            obj,
            server_side=True,
            certfile="./tls-cer/server-crt.pem",
            keyfile="./tls-cer/server-key.pem",
        )
        try:
            while serverSsl:
                message = serverSsl.read()
                data = obj.recv(1024)
                print (data);
                if message:
                    print(message)
                else:
                    print("Peguei foi nada")
        except Exception as ex:
            serverSsl.close()
            obj.close()

main();